BTrees/tests/common.py

```diff
@@ -208,6 +211,95 @@ def remove(i):
         self.assertTrue(100 in t)
         self.assertTrue(not read)
 
+    def test_impl_pickle(self):
+        # Issue #2
+        # Nothing we pickle should include the 'Py' suffix of
+        # implementation classes, and unpickling should give us
+        # back the best available type
+        import pickle
+        made_one = self._makeOne()
+
+        for proto in range(1, pickle.HIGHEST_PROTOCOL + 1):
+            dumped_str = pickle.dumps(made_one, proto)
+            self.assertTrue(b'Py' not in dumped_str, repr(dumped_str))
+
+            loaded_one = pickle.loads(dumped_str)
+
+            # If we're testing the pure-Python version, but we have the
+            # C extension available, then the loaded type will be the C
+            # extension but the made type will be the Python version.
+            # Otherwise, they match. (Note that if we don't have C extensions
+            # available, the __name__ will be altered to not have Py in it. See _fix_pickle)
+            if 'Py' in type(made_one).__name__:
+                self.assertTrue(type(loaded_one) is not type(made_one))
+            else:
+                self.assertTrue(type(loaded_one) is type(made_one) is self._getTargetClass(), (type(loaded_one), type(made_one), self._getTargetClass(), repr(dumped_str)))
```

Comment:
The problem with wrapping long lines in `assert` statements is that you lose the context in the traceback, making it harder to match up what failed :( . E.g., instead of something like
```
File "foo" 271
   assertTrue(a = b = c, (a, b, c))
Error: False is not true ("A", "B", "C")
```
You wind up with something like:
```
File "foo" 272
                     (a, b, c))
Error: False is not true ("A", "B", "C")
```